app/scripts/metamask-controller.js

```diff
@@ -788,6 +788,7 @@ module.exports = class MetamaskController extends EventEmitter {
       this.preferencesController.setAddresses(accounts)
       this.selectFirstIdentity()
       releaseLock()
+      await this.accountsController.fullUpdate()
```

Comment:
The fact that this is needed indicates that `accountsController` is probably not properly updating on update from `keyringController`